syntax = "proto3";

option go_package = "/hexcloud";

package endpoints.hexworld.hexcloud;

message Empty {}

message Hex {
    int64 X = 1; //
    int64 Y = 2;
    int64 Z = 3;
    string Type = 4;
    string Direction = 5;
}

message HexCubeList {
    repeated Hex hex = 1;
}

message HexCubeResponse {
    repeated Hex hc = 1;
}

message HexagonRingRequest {
    Hex hex = 1;
    int64 radius = 2;
    bool fill = 3;
}

message HexAmountResponse {
    int64 deleted = 1;
}

message Status {
    string msg = 1;
}

service HexagonService {
    rpc AddHexagons(HexCubeList) returns(Empty);
    rpc GetHexagonRing(HexagonRingRequest) returns (HexCubeResponse);
    rpc UpdateHexagon(HexCubeList) returns(HexAmountResponse);
    rpc DeleteHexagon(HexCubeList) returns(HexAmountResponse);
    rpc GetStatusServer(Empty) returns(Status);
    rpc GetStatusStorage(Empty) returns(Status);
    rpc GetStatusClients(Empty) returns(Status);
}